pennylane/hf/integrals.py

```diff
@@ -104,3 +104,179 @@ def contracted_norm(l, alpha, a):
+        lb (integer): angular momentum component for the second Gaussian function
+        ra (float): position component of the the first Gaussian function
+        rb (float): position component of the the second Gaussian function
+        alpha (array[float]): exponent of the first Gaussian function
+        beta (array[float]): exponent of the second Gaussian function
+        t (integer): number of nodes in the Hermite Gaussian
+
+    Returns:
+        array[float]: expansion coefficients for each Gaussian combination
+
+    **Example**
+
+    >>> la, lb = 0, 0
+    >>> ra, rb = 0.0, 0.0
+    >>> alpha = np.array([3.42525091])
+    >>> beta =  np.array([3.42525091])
+    >>> t = 0
+    >>> c = expansion(la, lb, ra, rb, alpha, beta, t)
+    >>> c
+    array([1.])
+    """
+    p = alpha + beta
+    q = alpha * beta / p
+    r = ra - rb
+
+    if la == lb == t == 0:
+        return anp.exp(-q * r ** 2)
+
+    if t < 0 or t > (la + lb):
+        return 0.0
+
+    if lb == 0:
+        return (
+            (1 / (2 * p)) * expansion(la - 1, lb, ra, rb, alpha, beta, t - 1)
+            - (q * r / alpha) * expansion(la - 1, lb, ra, rb, alpha, beta, t)
+            + (t + 1) * expansion(la - 1, lb, ra, rb, alpha, beta, t + 1)
+        )
+
+    return (
+        (1 / (2 * p)) * expansion(la, lb - 1, ra, rb, alpha, beta, t - 1)
+        + (q * r / beta) * expansion(la, lb - 1, ra, rb, alpha, beta, t)
+        + (t + 1) * expansion(la, lb - 1, ra, rb, alpha, beta, t + 1)
+    )
+
+
+def gaussian_overlap(la, lb, ra, rb, alpha, beta):
+    r"""Compute overlap integral for two primitive Gaussian functions.
+
+    The overlap integral between two Gaussian functions denoted by :math:`a` and :math:`b` can be
+    computed as [`Helgaker (1995) p803 <https://www.worldscientific.com/doi/abs/10.1142/9789812832115_0001>`_]:
+
+    .. math::
+
+        S_{ab} = E^{ij} E^{kl} E^{mn} \left (\frac{\pi}{p}  \right )^{3/2},
+
+    where :math:`E` is a coefficient that can be computed recursively, :math:`i-n` are the angular
+    momentum quantum numbers corresponding to different Cartesian components and :math:`p` is
+    computed from the exponents of the two Gaussian functions as :math:`p = \alpha + \beta`.
+
+    Args:
+        la (integer): angular momentum for the first Gaussian function
+        lb (integer): angular momentum for the second Gaussian function
+        ra (float): position vector of the the first Gaussian function
+        rb (float): position vector of the the second Gaussian function
+        alpha (array[float]): exponent of the first Gaussian function
+        beta (array[float]): exponent of the second Gaussian function
+
+    Returns:
+        array[float]: overlap integral between primitive Gaussian functions
+    """
+    p = alpha + beta
+    s = 1.0
+    for i in range(3):
+        s = s * anp.sqrt(anp.pi / p) * expansion(la[i], lb[i], ra[i], rb[i], alpha, beta, 0)
+    return s
+
+
+def generate_overlap(basis_a, basis_b):
```

Comment:
This function feels like the 'main' one in this module. Is it worth having an example?

Reply:
Added.